python/dgl/backend/pytorch/sparse.py

```diff
@@ -242,6 +246,18 @@ def backward(ctx, *dZ):
                 elif op == 'copy_lhs':
                     tpl_None = tuple([None] * len(Y))
                     dX = gspmm_hetero(g_rev, 'copy_lhs', 'sum', len(X), *tuple(dZ + tpl_None))
+            else:  # max/min
+                # TODO(Israt): dont just depend on the first etype
+                src_id, dst_id = gidx.metagraph.find_edge(0)
+                dX = tuple([th.zeros((X_shape[i][0],) + dZ[dst_id].shape[1:], dtype=dtype, device=device)
+                    if X[i] is not None else None for i in range(len(X))])
+                if op == 'mul':
+                    grad = _expand(Y, dZ.shape[1:]).gather(
+                        0, argY.long()) * dZ
+                    dX.scatter_add_(0, argX.long(), grad)
+                elif op in ['add', 'copy_lhs']:
+                    # TODO(Israt): argX.long()
```

Comment:
PyTorch's version uses `argX.long()` as shown in line 256. Not sure the reason behind it and whether we need to do something similar.

Reply:
PyTorch's version uses `long` because they only support 64-bit indices but not 32-bit indices.  If your `_update_grad_minmax_hetero` supports 64-bit indices then you don't need to convert it.

Reply:
Got it!